dartagnan/src/main/java/com/dat3m/dartagnan/program/Function.java

```diff
@@ -149,14 +152,34 @@ public void append(Event event){
         }
     }
 
+    public void append(List<Event> events) {
+        if (events.isEmpty()) {
+            return;
+        } else if (exit == null) {
+            append(events.get(0));
+            events = events.subList(1, events.size());
+        }
+        exit.insertAfter(events);
+    }
+
     public void updateExit(Event event){
+        Preconditions.checkArgument(event.getFunction() == this);
         exit = event;
         Event next;
         while((next = exit.getSuccessor()) != null){
             exit = next;
         }
     }
 
+    public void updateEntry(Event event) {
+        Preconditions.checkArgument(event.getFunction() == this);
+        entry = event;
+        Event next;
+        while((next = entry.getPredecessor()) != null){
+            entry = next;
+        }
+    }
+
```

Comment:
Isn't this actually updating the entry with the whole chain ending in `event`?

Reply:
It is a symmetric version to `updateExit`: it searches for the new entry event starting from `event` and going backwards (`getPredecessor`).
It's not a particularly nice method (just like `updateExit`) and should not really be called directly.